app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Types
+  class HmisSchema::QueryType < Types::BaseObject
+    # From generated QueryType:
+    # Add `node(id: ID!) and `nodes(ids: [ID!]!)`
+    include GraphQL::Types::Relay::HasNodeField
+    include GraphQL::Types::Relay::HasNodesField
+
+    field :projects, [Types::HmisSchema::Project], 'Get a list of projects' do
+      argument :project_types, [Types::HmisSchema::ProjectType], required: false
+    end
```

Comment:
Is this defining (and validating somewhere) the fields that should be present on `context` when it gets passed to the `projects` function in `app/graphql/types/hmis_schema/project.rb`? It feels slightly weird that the parameters are defined in a separate file from the function, but I'm sure we'll learn how to best organize this as the schema grows

Reply:
@gigxz I'm not sure I quite understand your question. `QueryType` defines the entry point for the GraphQL query... So the `field` says that `{ projects { ... } }` is a valid GraphQL query. The `argument` is an optional argument to the query, so it says that `{ projects(project_types: [...] { ... } }` where the contents of the array are of `ProjectType` is a valid query as well.

`ProjectType` is an enum type, and the GraphQL parse automatically translates these between the enum values and the Ruby values (e.g., when you put `project_types: [ES]` in the query, it will get passed to the Ruby parameter as `[1]`, and when the field `project_type` in `HmisSchema::Project` is returned, the 1 will appear in the JSON as `ES`.

Reply:
Thanks, that makes sense.

I think what I was getting at is that the "query parameters" for querying a Project are repeated in a bunch of places (here, line 20 below, and line 16 in `project.rb`). We only have one query param right now–– `project_types`-- but if that becomes a huge list it might be annoying to maintain it in several places. I think this is fine though!

Reply:
Yes, you are right, but, so far as I can tell, it is how it works. This declares the GraphQL schema, and the part below declares the query behavior. We could move the code from `Project`, but I think it is better to have it close the the types and this is just a forwarder - otherwise `QueryType` ends up with little bits of lots of different type models, but if there is a better way...